lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,451 @@
+
+    **Example requests**:
+
+        GET /api/courses/v0/{course_id}/
+
+        GET /api/courses/v0/{course_id}/?depth=2
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * content: When the depth parameter is used, a collection of child
+          course content entities, such as chapters, sequentials, and
+          components.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+
+    def get(self, request, course_id):
+        """
+        GET /api/courses/v0/{course_id}/
+        """
+        depth = int(request.QUERY_PARAMS.get('depth', 0))
+        # get_course_by_id raises an Http404 if the requested course is invalid
+        # Rather than catching it, we just let it bubble up
+        course_descriptor, course_key, _course_content = get_course(course_id, depth=depth)
+        if not course_descriptor:
+            raise Http404
+        response_data = _get_course_data(request, course_key, course_descriptor, depth)
+        return Response(response_data, status=status.HTTP_200_OK)
+
+
+class CourseGradedContent(PermissionMixin, generics.ListAPIView):
+    """
+    **Use Case**
+
+        Retrieves course graded content and problems.
+
+    **Example requests**:
+
+        GET /api/courses/v0/{course_id}/graded_content/
+
+    **Response Values**
+
+        * id: The ID of the content.
+
+        * name: The name of the content.
+
+        * format: The type of the content (e.g. Exam, Homework). Note: These values are course-dependent.
+          Do not make any assumptions based on assignment type.
+
+        * problems: {
+            * id: The ID of the problem.
+
+            * name: The name of the problem.
+        }
+    """
+
+    serializer_class = GradedContentSerializer
+    allow_empty = False
+
+    def _filter_problems(self, node):
+        """ Retrieve the problems from the node/tree. """
+
+        if node.category == u'problem':
+            return [node]
+
+        if not node.has_children:
+            return []
+
+        problems = []
+        for child in node.get_children():
+            problems += self._filter_problems(child)
+
+        return problems
+
+    def get_course_graded_content(self, course_key):
+        """
+        Retrieve course graded (e.g. homework, exams) from the module store.
+        """
+        _modulestore = modulestore()
+
+        # Ensure the course exists
+        if not _modulestore.get_course(course_key):
+            raise Http404
+
+        items = _modulestore.get_items(course_key, settings={'graded': True})
+
+        for item in items:
+            problems = self._filter_problems(item)
+            item.problems = problems
+
+        return items
```

Comment:
Note that this endpoint was originally called assignments. I don't want just problems. I need the actual assignment (e.g. homework, exam, etc.) so that I can build a linkage from assignment type to assignment to problem. I would prefer to not return the extraneous (for my use case) information in between the assignment node and the problem node.

I admit this is a fairly specific use case; however, it is preferable to the alternatives:
* return the complete course structure,
* return the full structure of the graded content nodes (which might as well be the complete course structure for some courses)
* return the top graded content node (what I call the assignment) and it's leaves (which are not guaranteed to be problems)

I'm open to suggestions.